devimint/src/main.rs

```diff
@@ -24,6 +24,47 @@ use tokio::fs;
 use tokio::net::TcpStream;
 use tracing::{debug, error, info, warn};
 
+struct Stats {
+    min: Duration,
+    avg: Duration,
+    median: Duration,
+    p90: Duration,
+    max: Duration,
+    sum: Duration,
+}
+
+impl std::fmt::Display for Stats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "min: {:.1}s", self.min.as_secs_f32())?;
+        write!(f, ", avg: {:.1}s", self.avg.as_secs_f32())?;
+        write!(f, ", median: {:.1}s", self.median.as_secs_f32())?;
+        write!(f, ", p90: {:.1}s", self.p90.as_secs_f32())?;
+        write!(f, ", max: {:.1}s", self.max.as_secs_f32())?;
+        write!(f, ", sum: {:.1}s", self.sum.as_secs_f32())?;
+        Ok(())
+    }
+}
+
+fn stats_for(mut v: Vec<Duration>) -> Stats {
+    assert!(!v.is_empty());
+    v.sort();
```

Comment:
nit: `sort_unstable` is slightly faster ;D . Not sure how big these are.

Reply:
20-30 elements